src/System.Drawing.Common/src/System/Drawing/Brush.cs

```diff
@@ -100,34 +60,13 @@ protected virtual void Dispose(bool disposing)
                     Debug.Assert(status == SafeNativeMethods.Gdip.Ok, "GDI+ returned an error status: " + status.ToString(CultureInfo.InvariantCulture));
 #endif
                 }
-                catch (Exception ex)
-                {
-                    if (ClientUtils.IsSecurityOrCriticalException(ex))
-                    {
-                        throw;
-                    }
-
-                    Debug.Fail("Exception thrown during Dispose: " + ex.ToString());
-                }
```

Comment:
The code for checking the exception is
```cs
public static bool IsCriticalException(Exception ex)
{
    return ex is NullReferenceException
            || ex is StackOverflowException
            || ex is OutOfMemoryException
            || ex is System.Threading.ThreadAbortException
            || ex is ExecutionEngineException
            || ex is IndexOutOfRangeException
            || ex is AccessViolationException;
}

public static bool IsSecurityOrCriticalException(Exception ex)
{
    return (ex is System.Security.SecurityException) || IsCriticalException(ex);
}
```

SecurityExceptions are not going to be thrown in .NET Core, as the coreclr doesn't support these security features.

The only other option that can be thrown from a delete is an AccessViolationException, which can occur in the following code example:
```
[Fact]
public void SetNativeBrush_Brush_Success()
{
    using (var brush = new SubBrush())
    {
        brush.PublicSetNativeBrush((IntPtr)10);
    }
}

private class SubBrush : Brush
{
    public override object Clone() => this;
    public void PublicSetNativeBrush(IntPtr brush) => SetNativeBrush(brush);
}
```

It's my understanding that GDI+ can't throw CLR exceptions, so only an AV can occur, which will already be handled.